tests/test_stac_io.py

```diff
@@ -142,3 +143,10 @@ def test_retry_stac_io_404() -> None:
             "https://planetarycomputer.microsoft.com"
             "/api/stac/v1/collections/not-a-collection-id"
         )
+
+
+def test_save_http_href_errors(tmp_path: Path) -> None:
+    catalog = pystac.Catalog(id="test-catalog", description="")
+    catalog.set_self_href("http://pystac.test/catalog.json")
+    with pytest.raises(NotImplementedError):
```

Comment:
nit: ensure you are getting the error you expect by testing that the message matches.

```suggestion
    with pytest.raises(NotImplementedError, match="cannot write to urls"):
```

Reply:
Thanks, I'm going to keep as is -- I'm a little 👎🏼 matching on error strings, since (IMO) we should be able to change the message to be more informative w/o breaking the test. No one _should_ be depending on certain text in the error message to drive downstream behavior, so it's not something that needs to be validated in CI (again, IMO).